sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetEncodingSuite.scala

```diff
@@ -78,4 +78,30 @@ class ParquetEncodingSuite extends ParquetCompatibilityTest with SharedSQLContex
       }}
     }
   }
+
+  test("Read row group containing both dictionary and plain encoded pages") {
+    spark.conf.set("parquet.dictionary.page.size", "2048")
+    spark.conf.set("parquet.page.size", "4096")
```

Comment:
Nit: You can use the following constants instead of hard code the key strings:

- `ParquetOutputFormat.DICTIONARY_PAGE_SIZE`
- `ParquetOutputFormat.PAGE_SIZE`